server/src/main/java/io/crate/exceptions/MissingParametersException.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.exceptions;
+
+public class MissingParametersException extends RuntimeException implements CrateException {
```

Comment:
if we introduce a new `CrateException` do we need to have a unique error code? Do we also have to reflect changes here https://github.com/crate/crate/blob/ad02ff5f40bc853937b15eef6e9dcae5543838f9/server/src/main/java/io/crate/exceptions/SQLExceptions.java#L144-L163 ?

Do we need to update http doc https://github.com/crate/crate/blob/ad02ff5f40bc853937b15eef6e9dcae5543838f9/docs/interfaces/http.rst#id10 

It seems like a user-facing change. Were you planning to backport it as well?

Reply:
Could we just go with "InvalidArgumentException" for now? I had in mind to restructure the exception handling a bit to make it easier to return the proper error codes in the PostgreSQL wire protocol, so it would be helpful if we didn't add more `CrateException` instances for now.

Reply:
As commented I suggest to delete this and let `BlobStoreCredentialsException` extend from `ValidationExecption`.
But still this would introduce a new error code and thus user facing change.
I think this could be acceptable for a backport as it should not break anything.

Reply:
If we use an `ValidationExecption` then we cannot add a designated error code. We reserved all 10 error code on the 400 already https://crate.io/docs/crate/reference/en/latest/interfaces/http.html. We could either fallback to 4000 or move it to 401X (Unauthorized).

Reply:
> If we use an `ValidationExecption` then we cannot add a designated error code. We reserved all 10 error code on the 400 already https://crate.io/docs/crate/reference/en/latest/interfaces/http.html. We could either fallback to 4000 or move it to 401X (Unauthorized).

I'd go with 4000. 

For 401X we need to be careful. We need to distinguish between unauthorized to execute certain SQL statements (permissions related to executing CrateDB stuff), and unauthorized to use a other third-party resource.
We also need to be careful how clients would map these errors to exceptions.